components/raftstore/src/store/worker/pd.rs

```diff
@@ -2162,4 +2188,39 @@ mod tests {
             assert!(*region_cpu_records.get(&region_id).unwrap_or(&0) > 0)
         }
     }
+
+    #[test]
+    fn test_collect_used_size() {
+        let tmp_path = tempfile::Builder::new()
+            .prefix("test_cluster")
+            .tempdir()
+            .unwrap();
+        let engines = engine_test::new_temp_engine(&tmp_path);
+        let store_info = StoreInfo {
+            kv_engine: engines.kv,
+            raft_engine: engines.raft,
+            capacity: 0,
+        };
+
+        let tmp_mgr_path = tmp_path.path().join(std::path::Path::new("snap"));
+        let snap_mgr = SnapManager::new(tmp_mgr_path.to_str().unwrap());
+        let mut stats = pdpb::StoreStats::default();
+
+        let fp = "get_total_snap_size";
+
+        fail::cfg(fp, "return(0)").unwrap();
+        stats = collect_used_size(&snap_mgr, &store_info, stats);
+        let used_size0 = stats.used_size;
+
+        fail::cfg(fp, "return(1024)").unwrap();
+        stats = collect_used_size(&snap_mgr, &store_info, stats);
+        let used_size1 = stats.used_size;
+        assert_eq!(used_size1, used_size0 + 1024);
+
+        fail::cfg(fp, "return").unwrap();
+        stats = collect_used_size(&snap_mgr, &store_info, stats);
+        assert_eq!(stats.used_size, used_size1);
+
+        fail::remove(fp);
```

Comment:
failpoint cases can't be run concurrently and there is no such guarantee when running test cases locally. So this may make other cases fail randomly.

Reply:
So failpoint cases should be placed in `tests/failpoints`, then they will run serially by a custom test frameworks.
Am I correct ?

Reply:
OK, thanks~